powerlevel9k.zsh-theme

```diff
@@ -610,6 +609,28 @@ prompt_dir() {
           current_path=$(truncatePathFromRight "$(pwd | sed -e "s,^$HOME,~,")" )
         fi
       ;;
+      truncate_with_folder_marker)
+        local last_marked_folder marked_folder zero
+        set_default POWERLEVEL9K_SHORTEN_FOLDER_MARKER ".shorten_folder_marker"
+
+        for marked_folder in $(upsearch $POWERLEVEL9K_SHORTEN_FOLDER_MARKER); do
+          if [[ "$marked_folder" == "/" ]]; then
+            # If we reached root folder, stop upsearch.
+            current_path="/"
+          elif [[ "$marked_folder" == "$HOME" ]]; then
+            # If we reached home folder, stop upsearch.
+            current_path="~"
+          elif [[ "${marked_folder%/*}" == $last_marked_folder ]]; then
+            current_path="${current_path%/}/${marked_folder##*/}"
+          else
+            current_path="${current_path%/}/$POWERLEVEL9K_SHORTEN_DELIMITER/${marked_folder##*/}"
+          fi
+          last_marked_folder=$marked_folder
+        done
+
+        zero='%([BSUbfksu]|([FB]|){*})'
+        current_path=$current_path${PWD:${#${(S%%)last_marked_folder//$~zero/}}}
+      ;;
```

Comment:
@dritter @BenoitAverty - What are your thoughts on Benoit's comment, here? Do you want to iterate on this function or should we merge as-is?